### PR TITLE
Bump android gradle plugin to 3.6.1

### PR DESCRIPTION
**Pull request by raviagarwal7**
- Resolves https://github.com/uber/okbuck/issues/840
- Update these libraries to make the build to work
```
-        androidPlugin      : "3.2.0",
-        androidTools       : "26.2.0",
-        butterKnife        : "10.0.0",
-        dagger             : "2.16",
-        kotlin             : "1.3.21",
+        androidPlugin      : "3.6.1",
+        androidTools       : "26.6.1",
+        butterKnife        : "10.2.1",
+        dagger             : "2.28.1",
+        kotlin             : "1.3.72",
```
- Bump robolectric to 4.3.1
- Materialize kotlin_home target as part of buckw so the build works with oss buck.

### CODE REVIEW

File: dependencies.gradle [hunk 1]
```diff
@@ -17,11 +17,11 @@ def exclude = { dep, String... excludes  ->
 }
 
 def versions = [
-        androidPlugin      : "3.2.0",
+        androidPlugin      : "3.6.1",
         androidTools       : "26.2.0",
```
Comment (kageiit) [inline dependencies.gradle]: need to bump tools also to 26.6.1

File: buildSrc/build.gradle [hunk 2]
```diff
@@ -101,7 +101,7 @@ task sourcesJar(type: Jar) {
     classifier = "sources"
 }
 
-def publishVersion = "0.50.10"
+def publishVersion = "0.50.11-LOCAL"
```
Comment (kageiit) [inline buildSrc/build.gradle:104]: undo

File: dependencies.gradle [hunk 3]
```diff
@@ -82,6 +82,7 @@ def external = [
         kotlinExtension : "org.jetbrains.kotlin:kotlin-android-extensions:${versions.kotlin}",
         kotlinStdlib    : "org.jetbrains.kotlin:kotlin-stdlib:${versions.kotlin}",
         leakCanary      : "com.squareup.leakcanary:leakcanary-android:${versions.leakCanary}",
+        sqldelight      : "com.squareup.sqldelight:android-driver:1.4.0",
```
Comment (kageiit) [inline dependencies.gradle:85]: move 1.4.0 to versions

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,5 +1,6 @@
+#Wed Jul 15 22:50:36 CEST 2020
```
Comment (kageiit) [inline gradle/wrapper/gradle-wrapper.properties:1]: remove timestamp

**Comment by raviagarwal7** [comment]
```
Warning: com.google.android.material.tabs.TabLayout$TabView: can't find referenced class androidx.appcompat.content.res.AppCompatResources
Warning: com.google.android.material.tabs.TabLayout$TabView: can't find referenced class androidx.appcompat.content.res.AppCompatResources
Warning: com.google.android.material.textfield.TextInputLayout: can't find referenced class androidx.appcompat.widget.DrawableUtils
Warning: com.google.android.material.textfield.TextInputLayout: can't find referenced class androidx.appcompat.widget.DrawableUtils
Warning: com.google.android.material.textfield.TextInputLayout: can't find referenced class androidx.appcompat.content.res.AppCompatResources
Warning: com.google.android.material.textfield.TextInputLayout: can't find referenced class androidx.appcompat.content.res.AppCompatResources
Warning: there were 192 unresolved references to classes or interfaces.
         You may need to add missing library jars or update their versions.
         If your code works fine without the missing classes, you can suppress
         the warnings with '-dontwarn' options.
         (http://proguard.sourceforge.net/manual/troubleshooting.html#unresolvedclass)
Warning: there were 3 unresolved references to program class members.
         Your input classes appear to be inconsistent.
         You may need to recompile the code.
         (http://proguard.sourceforge.net/manual/troubleshooting.html#unresolvedprogramclassmember)
java.io.IOException: Please correct the above warnings first.
	at proguard.Initializer.execute(Initializer.java:485)
	at proguard.ProGuard.initialize(ProGuard.java:285)
	at proguard.ProGuard.execute(ProGuard.java:114)
	at proguard.ProGuard.main(ProGuard.java:594)

    When running <proguard_obfuscation>.
    When building rule //another-app:bin_release#class_file_to_dex_processing.
```

The build now fails at the proguard step. @kageiit do you know what might be up?

**Comment by kageiit** [comment]
> ```
> Warning: com.google.android.material.tabs.TabLayout$TabView: can't find referenced class androidx.appcompat.content.res.AppCompatResources
> Warning: com.google.android.material.tabs.TabLayout$TabView: can't find referenced class androidx.appcompat.content.res.AppCompatResources
> Warning: com.google.android.material.textfield.TextInputLayout: can't find referenced class androidx.appcompat.widget.DrawableUtils
> Warning: com.google.android.material.textfield.TextInputLayout: can't find referenced class androidx.appcompat.widget.DrawableUtils
> Warning: com.google.android.material.textfield.TextInputLayout: can't find referenced class androidx.appcompat.content.res.AppCompatResources
> Warning: com.google.android.material.textfield.TextInputLayout: can't find referenced class androidx.appcompat.content.res.AppCompatResources
> Warning: there were 192 unresolved references to classes or interfaces.
>          You may need to add missing library jars or update their versions.
>          If your code works fine without the missing classes, you can suppress
>          the warnings with '-dontwarn' options.
>          (http://proguard.sourceforge.net/manual/troubleshooting.html#unresolvedclass)
> Warning: there were 3 unresolved references to program class members.
>          Your input classes appear to be inconsistent.
>          You may need to recompile the code.
>          (http://proguard.sourceforge.net/manual/troubleshooting.html#unresolvedprogramclassmember)
> java.io.IOException: Please correct the above warnings first.
> 	at proguard.Initializer.execute(Initializer.java:485)
> 	at proguard.ProGuard.initialize(ProGuard.java:285)
> 	at proguard.ProGuard.execute(ProGuard.java:114)
> 	at proguard.ProGuard.main(ProGuard.java:594)
> 
>     When running <proguard_obfuscation>.
>     When building rule //another-app:bin_release#class_file_to_dex_processing.
> ```
> 
> The build now fails at the proguard step. @kageiit do you know what might be up?

@kurtisnelson bumped proguard in a recent PR. try to revert those changes and check?

**Comment by kurtisnelson** [comment]
I didn't actually bump the version though, just the handling logic.

**Pull request merged**